examples/animation/main.rs

```diff
@@ -190,19 +190,21 @@ fn main() -> amethyst::Result<()> {
     let display_config_path = app_root.join("examples/animation/resources/display_config.ron");
     let resources = app_root.join("examples/assets/");
 
+    let mut world = World::new();
+
     let game_data = GameDataBuilder::default()
```

Comment:
Could we instead have GameDataBuilder optionally have a constructor where it receives a user's World (constructing one with ::new() by default) and add an argument to SystemBundle::build for receiving the reference to that instance? The costs would be offloaded to changing the API for SystemBundle but I think this solution would be better in the long term since it's a utility interface for building dispatchers anyways.

Then, the calls to ::with_bundle here would not take the &mut world. The problem would arise when users have simple systems which take a reference to world for initialization alongside bundles in their gamedatabuilder, since that would create two simultaneous mutable borrows on world. Such systems might need to be done in a GameDataBuilder method that takes an anonymous function, with the argument being the reference to that World.

Reply:
Doesn't work, because the user is also creating systems that require &mut world access for their new() function. :(

Reply:
I said that :P 

> The problem would arise when users have simple systems which take a reference to world for initialization alongside bundles in their gamedatabuilder, since that would create two simultaneous mutable borrows on world. Such systems might need to be done in a GameDataBuilder method that takes an anonymous function, with the argument being the reference to that World.

Reply:
ah right :P

Reply:
Closing this. If I figure out a nice way to do it, I'll open an issue dedicated for it :P